### PR TITLE
ENH Adds items to dataframe

### PR DESCRIPTION
**Pull request by thomasjpfan**
Fixes https://github.com/dask/dask/issues/6501

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -4148,6 +4148,11 @@ def itertuples(self, index=True, name="Pandas"):
             for row in df.itertuples(index=index, name=name):
                 yield row
 
+    @derived_from(pd.DataFrame)
+    def items(self):
+        for key in self.columns:
+            yield key, self[key]
```
Comment (jsignell) [inline dask/dataframe/core.py:4154]: Tiny comment: maybe use `label` and `content` like pandas does? 

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.items.html

**Comment by jsignell** [comment]
Awesome!

**Comment by thomasjpfan** [comment]
I just remembered that pandas can have non unique column names. This PR was updated to support this.

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -4148,6 +4148,11 @@ def itertuples(self, index=True, name="Pandas"):
             for row in df.itertuples(index=index, name=name):
                 yield row
 
+    @derived_from(pd.DataFrame)
+    def items(self):
+        for col_idx, label in enumerate(self.columns):
+            yield label, self.iloc[:, col_idx]
```
Comment (jsignell) [inline dask/dataframe/core.py:4154]: Does this work with MultiIndexes?

Comment (jsignell) [reply dask/dataframe/core.py:4154]: Seems to locally.

**Comment by jsignell** [comment]
I'm planning on merging when green unless @TomAugspurger has qualms.

**Comment by thomasjpfan** [comment]
Updated PR with test for multiindex.

**Pull request merged**